core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -905,22 +905,22 @@ class KafkaApis(val requestChannel: RequestChannel,
         if (header.apiVersion == 0) {
           val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
             .partition(authorizeTopicDescribe)
+          // authorizedAndExistingPartitions: authorized partitions that actually exist in the cluster, and need their offset retrieved
+          // nonExistingPartitions: partitions that do not exist in the cluster, and should get an unknown partition error
+          val (authorizedAndExistingPartitions, nonExistingPartitions) =
```

Comment:
Some of the redundance between this branch and the one below is annoying. Is there any way to use an inner def to reduce the duplication? How about something like this?

```scala
def authorizeAndCollectOffsets(authorizedAndExistingPartitions => (error, authorizedAndExistingPartitionData)): OffsetFetchResponse
```

(With the corresponding types filled in.) Would that work?